File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-205812-95418/output/project/src/customerimporter/old_zip_code.py
from customerimporter.zip_code import ZipCode


class OldZipCode(ZipCode):

    # error messages

    @classmethod
    def invalid_code_error_description(cls):
        return "Invalid old zipcode"

    # type

    @classmethod
    def is_for(cls, a_zip_code):
        return a_zip_code[0].isdigit()

    # code - validation

    @classmethod
    def assert_is_valid_code(cls, a_zip_code):
        cls.assert_valid_code(a_zip_code.isdigit())
        imported_zip_code = int(a_zip_code)
        cls.assert_valid_code(imported_zip_code >= 1000 and imported_zip_code <= 9999)

    # initialization

    def __init__(self, a_zip_code):
        super().__init__(int(a_zip_code))

    # code

    def new_zip_code_if_none(self, a_none_block):
        return a_none_block()

    def old_zip_code_if_none(self, a_none_block):
        return self._code

    def is_new(self):
        return False

    def is_old(self):
        return True

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-204311-76559/analysis/acceptance-project/src/main/java/customerimporter/OldZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class OldZipCode extends ZipCode {

    private static final int MINIMUM_NUMBER = 1000;
    private static final int MAXIMUM_NUMBER = 9999;

    private final int number;

    // instance creation

    public static boolean canBeCreatedFrom(String aZipCode) {
        return Character.isDigit(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidZipCodeErrorDescription() {
        return "Invalid old zipcode";
    }

    // number - a new zip code carries an old zip code number, so both validate it the same way,
    // each one reporting its own error

    static int assertIsValidNumber(String aNumber, String anErrorDescription) {
        return Validation.assertIsNumberBetween(aNumber, MINIMUM_NUMBER, MAXIMUM_NUMBER, anErrorDescription);
    }

    // initialization

    OldZipCode(String aZipCode) {
        number = assertIsValidNumber(aZipCode, invalidZipCodeErrorDescription());
    }

    // zip code

    public Object value() {
        return number;
    }

    // testing

    public boolean isNew() {
        return false;
    }

    public boolean isOld() {
        return true;
    }

    // values

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return number;
    }
}
